File: linkedList/reverseLinkedList.js
// https://leetcode.com/problems/reverse-linked-list/
const reverseList = (head) => {
  let [prev, current] = [null, head];
  while (current) {
    [current.next, prev, current] = [prev, current, current.next];
  }
  return prev;
};

File: linkedList/reverseLinkedList.py
# https://leetcode.com/problems/reverse-linked-list/
class Solution:
    def reverseList(self, head):
        prev = None
        while head:
            curr = head
            head = head.next
            curr.next = prev
            prev = curr
        return prev


class Solution:
    def reverseList(self, head):
        return self._reverse(head)

    def _reverse(self, node, prev=None):
        if not node:
            return prev
        n = node.next
        node.next = prev
        return self._reverse(n, node)
